musl libc doesn't support execinfo.h. Make sure the feature test macro
checks for __GLIBC__ instead of __linux__ to account for that.

diff -upr verilator-5.004.orig/include/verilated.cpp verilator-5.004/include/verilated.cpp
--- verilator-5.004/include/verilated.cpp	2023-01-05 18:27:05.479605957 +0100
+++ verilator-5.004.orig/include/verilated.cpp	2022-12-15 03:59:58.000000000 +0100
@@ -67,7 +67,7 @@
 #if defined(_WIN32) || defined(__MINGW32__)
 # include <direct.h>  // mkdir
 #endif
-#ifdef __linux__
+#ifdef __GLIBC__
 # include <execinfo.h>
 # define _VL_HAVE_STACKTRACE
 #endif
